plugins/catalog-backend-module-msgraph/api-report.md

```diff
@@ -233,6 +238,11 @@ export type OrganizationTransformer = (
   organization: MicrosoftGraph.Organization,
 ) => Promise<GroupEntity | undefined>;
 
+// @public
+export type ProviderTransformer = (
+  provider: MicrosoftGraphProviderConfig,
```

Comment:
Wonder if we should exclude a couple of the fields here that don't make sense to change, like the schedule? Then again the ID should be changed but is quite useful to have. Might perhaps be enough with some light docs on which fields can't be changed by the transformer?

Reply:
I thought about trying to sanitize fields like that, but I didn't for two reasons

- This is going to create more maintenance or weird scenarios as other options are added/removed in the future
- I think we can update the documentation to explicitly state the risks of modifying the config on the fly and leave it up to the user (in the end, the user can also screw up the static config and as written, the same checks will be applied by the `readMicrosoftGraphOrg()` for both the static and dynamic config

thoughts on that @Rugvip ?

Reply:
I think it's best to have a fairly well defined contract so that the user doesn't have to keep guessing or read the source code. i.e. we should commit to keeping one set of fields configurable on the fly and make sure the code enables that, and the other fields are not supported. In practice I wouldn't be too worried, I'm thinking we could just document something like "Transforming fields that are not used for each scheduled ingestion will have no effect, such as `id` and `schedule`."

Reply:
Ah ok - so, you aren't thinking any code changes are required to protect `id` and `schedule` (since those cannot be affected)?

Reply:
Not particularly, as long as it doesn't end up breaking things if users modify them and the behavior is documented in some form. It's kinda tricky to handle this at the TypeScript level too, except that excluding `schedule` might be worthwhile

Reply:
do you have recommendations on how to exclude `schedule` - the thoughts I have had so far seem cumbersome (e.g., interface with specific options, etc)